enhancements/20250407-cpu-model-features-selection.md

```diff
@@ -0,0 +1,385 @@
+
+- Scenario 4: Person requiring a moving to a new location  
+  A person requires an object with a specific color and shape (e.g., Red Circle) and is moving to a new location.  
+  The new location must provide an object that matches the color and shape (e.g., a Red Circle or a Red-Blue Circle).
+
+A node can provide objects with different shapes and colors. For example, a Node might provide a "Red Square", a "Blue Circle", and a "Red-Blue Circle". 
+
+A VM is like a person who requires different objects. 
+
+Migration is like moving a person with colored shape requirements to a new location that can fulfill those requirements. The destination must provide objects that meet the person's color and shape needs.
+
+You can treat shapes as `cpu-model.node.kubevirt.io/{cpu model}` and colors as `cpu-feature.node.kubevirt.io/{cpu feature}`. For more details, please check the Node section of this enhancement.
+
+## Proposal
+
+Due to multiple nodes, we can't show a big matrix for all the CPU models and features. Instead, we'll provide a dropdown selection menu with colored mark.
+
+### User Stories
+
+#### Story 1
+
+I have multiple nodes that have a common CPU model called `Nehalem`. For some reason, I'll need to migrate my VMs to other nodes without manually shutting down. To ensure compatibility, I create my VMs with `Nehalem` as the CPU model.
+
+### User Experience In Detail
+
+In general, there are two ways to migrate VMs between nodes.
+
+- Use a common CPU model
+- Use a Harvester special defined annotation
+
+Before users use the feature, we need to ensure they know what the difference is between these two ways.
+
+- If they know later joined nodes still have common CPU models, they can select a common CPU model for further migration.
+
+In either case, we need to add a ["learn more for migration"](https://docs.harvesterhci.io/v1.5/vm/live-migration/#how-migration-works) link when users try to customize the CPU model and feature.
+
+### API changes
+
+No.
+
+## Design
+
+The CPU models are like this:
+
+- Node-1 cpuModel: IvyBridge, Penryn
+- Node-2 cpuModel: IvyBridge, Westmere
+- Node-3 cpuModel: IvyBridge, SandyBridge
+- Node-4 cpuModel: IvyBridge, Westmere
+
+We'll only show the common one. These are just examples. The real one will match our Harvester GUI style.
+
+![](20250407-cpu-model-features-selection/01-custom-model.png)
+
+![](20250407-cpu-model-features-selection/02-custom-model.png)
+
+
+
+### Implementation Overview
+
+#### Frontend
+
+We'll have two ways to configure this.
+
+- Per-VM CPU model/features while creating the VM
+- Global VM CPU model/features in settings
+
+Frontend needs to create a new tab in the VM creation page and provide a dropdown selection menu for models and an input box for features. This selection is also available in the VM template page and global settings. Then, calculate the common CPU models across all nodes. 
+
+- If users select the common one, we'll show the option with green mark.
```

Comment:
Is the idea to identify the common one by examining the labels of all the nodes in the cluster? What if there isn't a common one among _all_ the nodes, but only a common one among _some_ of the nodes? This computation also needs to happen every time the user navigates to the VM page whether to create a new VM or update an existing one, right?